### DISCUSSION TITLE
Grab header value without header name

### DISCUSSION THREAD

**Discussion by bublikOff** [actix/actix-web#3104] [General]
I'm trying to grab auth header value but it returns full header value (name and value) ... are there built-in method to grab only value?
``
let auth = request.headers().get(ActixHttpHeader::AUTHORIZATION).unwrap().to_str().unwrap();
``

**Comment by robjtede** [comment]
Can you show a complete program demonstrating this behavior?

**Comment by bublikOff** [comment]
For the moment I do use this construction

```
    let token = match request.headers().get(ActixHttpHeader::AUTHORIZATION).map(ActixHttpHeader::HeaderValue::to_str)
    {
        Some(Ok(header)) => header.split_terminator(":").last().unwrap().trim().strip_prefix("Bearer").unwrap().trim(),
        None => ""
    };

```

but as I mentioned before when I trying to get header it returned with header name ... 
As for me its quite strange that HeaderValue also contains HeaderName

As an eg lets take following code as an example

```
let auth = request.headers().get(ActixHttpHeader::AUTHORIZATION).unwrap().to_str().unwrap();
println!("auth: {}", auth);


```
it will printout something like that

`auth: Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9. ......................`


Am a new one in rust and actix and maybe Ive missed something ... Ive tried  to google methods how to get header ... without header value but had no luck

**Reply by dns2utf8-tipi** [reply]
See on the wire if the client sends something like this:
```
authorization: Authorization: Bearser ey....
```

It could be that the client is just sending too much.
Anyways, you will have to split the string at `"Bearer "` and then take the JWT token out.

Or use one of the crates: https://crates.io/search?q=jwt%20actix
